Research/lva-node-app/app.js

```diff
@@ -0,0 +1,331 @@
+ * @requires module:body-parser 
+ * @requires module:express
+ * @requires module:iot-hub-connection-string
+ */
+const express = require('express');
+const WebSocket = require('ws');
+const EventHubReader = require('./eventhubreader.js');
+const iothub = require('azure-iothub');
+const bodyParser = require('body-parser');
+const { request, response } = require('express');
+const { convertIotHubToEventHubsConnectionString } = require('./iot-hub-connection-string.js');
+
+const app = express();
+const server= require('http').createServer(app);
+const wss = new WebSocket.Server({server});
+const Client = iothub.Client;
+
+app.use(bodyParser.urlencoded({extended: false}));
+app.use(bodyParser.json());
+app.use(express.static('public'))
+app.use(express.static('public/pages'))
+
+/**
+* @global
+*/
+global.DEVICE_ID="";
+global.MODULE_ID="lvaEdge";
+global.IOTHUB_CONNECTION_STRING="";
+global.IOTHUB_ENDPOINT="";
+
+var graphInstances={};
+var graphTopologies={};
+var cameras={};
+var eventHubReader;
+
+const PORT=5000;
+
+/**
+ * For all following declarations of function(req, res):
+ * @param {XMLHttpRequest} req - request object
+ * @param {XMLHttpRequestResponseType} res - request response object
+ */
+
+// render the html pages. the '/<name>' is what the url extension will look like in the browser
+app.get('/', function (req, res) 
+{
+    res.render('index.html')
+})
+app.get('/mediagraph', function (req, res) 
+{
+    res.render('mediagraph.html')
+})
+app.get('/configuration', function (req, res) 
+{
+    res.render('configuration.html')
+})
+app.get('/cameras', function (req, res) 
+{
+    res.render('cameras.html')
+})
+app.get('/output', function (req, res) 
+{
+    res.render('output.html')
+})
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+// handle GET requests
+
+/**
+* send client global variables upon request 
+*/
+app.get('/globals', function (req, res) 
+{
+  console.log("User requested globals from server");
+  let globals =
+  {
+    graphInstances: graphInstances,
+    graphTopologies: graphTopologies,
+    cameras: cameras
+  };
+  res.json(globals);
+})
+
+/**
+* update global variables when sent by user (user called setGlobals)
+* modifying existing variables, thus using a PUT request
+*/
+app.put('/globals', function (req, res)
+{
+  console.log("User setting server's globals");
+  graphInstances=req.body.graphInstances;
+  graphTopologies=req.body.graphTopologies;
+  cameras=req.body.cameras;
+  res.end();
+})
+
+/**
+* start sending live stream hub messages to client via websocket 
+* @returns {void}
+*/
+app.get('/hubMessages', function(req, res)
+{
+  if (validCredentialsAreSet(res))
+  {
+    receiveHubMessages();
+  } 
+  res.end();
+})
+
+/**
+ * stop receiving messages from iot hub
+ * @returns {void}
+ */
+app.get('/stopMessages', function(req, res)
+{
+  eventHubReader.stopReadMessage();
```

Comment:
Looks like I can't unresolve things so making a new comment. I don't see how the eventHubReader issues is esolved? If eventHubReader isn't set in app.js this still seems problematic... what is the change in eventHubReader as well that makes this work?